server/container/util/src/main/java/org/apache/james/util/concurrency/ConcurrentTestRunner.java

```diff
@@ -48,14 +54,19 @@ public ConcurrentRunnableTask(int threadNumber, BiConsumer biConsumer) {
 
         @Override
         public void run() {
+            exception = null;
             countDownLatch.countDown();
             for (int i = 0; i < operationCount; i++) {
                 try {
                     biConsumer.consume(threadNumber, i);
                 } catch (Exception e) {
                     LOGGER.error("Error caught during concurrent testing", e);
+                    exception = e;
```

Comment:
Could be better to throw all catched exception by wrapping them in a meta exception something like. MultipleException lol.
Otherwise we could give a return type of run returning a list of potential exception